pkg/certificate/providers/types.go

```diff
@@ -81,7 +81,8 @@ type MRCCompatClient struct {
 
 // MRCProviderGenerator knows how to convert a given MRC to its appropriate provider.
 type MRCProviderGenerator struct {
-	kubeClient kubernetes.Interface
+	compute.Interface
```

Comment:
There's no need to embed this interface, since we don't have callers accessing it directly (we're not leveraging the passthrough features of embedded structs).